base/thread_pool_body.hpp

```diff
@@ -63,11 +61,13 @@ void ThreadPool<T>::DequeueCallAndExecute() {
     // Wait until either the queue contains an element or this class is shutting
     // down.
     {
-      std::unique_lock<std::mutex> l(lock_);
-      has_calls_or_shutdown_.wait(l,
-                                  [this] {
-                                      return shutdown_ || !calls_.empty();
-                                  });
+      absl::MutexLock l(&lock_);
+
+      auto const has_calls_or_shutdown = [this] {
+        return shutdown_ || !calls_.empty();
+      };
+      lock_.Await(absl::Condition(&has_calls_or_shutdown));
```

Comment:
We don't need to name the λ.

We need to use unary `+` to make it the argument directly, because C++: https://github.com/abseil/abseil-cpp/blob/a4c3fff/absl/synchronization/mutex.h#L626-L639.

Reply:
As discussed, this doesn't work because of the capture.

Also, having a name is nice.